### PR TITLE
BZ2141719 ahv_internal_debug virt-who config support

### PR DESCRIPTION
**Pull request by yanpliu**
Realize BZ2141719 ahv_internal_debug virt-who config support for virt-who plugin

Nutanix UI Cases:PASS
```
(robottelo_vv) [yanpliu@yanpliu robottelo]$ pytest ./tests/foreman/virtwho/ui/test_nutanix.py -k test_positive_ahv_internal_debug_option
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.10.9, pytest-7.2.0, pluggy-1.0.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/yanpliu/gitrepository/robottelo, configfile: pyproject.toml
plugins: cov-3.0.0, xdist-3.1.0, services-2.2.1, mock-3.10.0, reportportal-5.1.3, ibutsu-2.2.4
collected 7 items / 13 deselected / -6 selected                                                                                                                                                                   

tests/foreman/virtwho/ui/test_nutanix.py .                                                                                                                                                                  [100%]

================================================================================================ warnings summary =================================================================================================

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================================ 1 passed, 13 deselected, 4 warnings in 132.26s (0:02:12) =============================================================================
```

Nutanix  API Cases:PASS
```
(robottelo_vv) [yanpliu@yanpliu robottelo]$ pytest ./tests/foreman/virtwho/api/test_nutanix.py -k test_positive_ahv_internal_debug_option
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.10.9, pytest-7.2.0, pluggy-1.0.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/yanpliu/gitrepository/robottelo, configfile: pyproject.toml
plugins: cov-3.0.0, xdist-3.1.0, services-2.2.1, mock-3.10.0, reportportal-5.1.3, ibutsu-2.2.4
collected 7 items / 13 deselected / -6 selected                                                                                                                                                                   

tests/foreman/virtwho/api/test_nutanix.py .                                                                                                                                                                 [100%]

================================================================================================ warnings summary =================================================================================================
-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================================= 1 passed, 13 deselected, 5 warnings in 94.71s (0:01:34) =============================================================================

```

Nutanix CLI Cases:PASS
```
(robottelo_vv) [yanpliu@yanpliu robottelo]$ pytest ./tests/foreman/virtwho/cli/test_nutanix.py -k test_positive_ahv_internal_debug_option
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.10.9, pytest-7.2.0, pluggy-1.0.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/yanpliu/gitrepository/robottelo, configfile: pyproject.toml
plugins: cov-3.0.0, xdist-3.1.0, services-2.2.1, mock-3.10.0, reportportal-5.1.3, ibutsu-2.2.4
collected 7 items / 13 deselected / -6 selected                                                                                                                                                                   

tests/foreman/virtwho/cli/test_nutanix.py .                                                                                                                                                                 [100%]

================================================================================================ warnings summary =================================================================================================

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================================= 1 passed, 13 deselected, 1 warning in 95.89s (0:01:35) ==============================================================================
(robottelo_vv) [yanpliu@yanpliu robottelo]$ 

```

### CODE REVIEW

**Comment by yanpliu** [comment]
Related
ahv_internal_debug option support for virt-who configuration #889
ahv_internal_debug virt-who config support #803

File: robottelo/utils/virtwho.py [hunk 1]
```diff
@@ -214,6 +214,21 @@ def get_rhsm_log():
...
 
+def check_message_in_rhsm_log(message):
+    """Check the message exist in /var/log/rhsm/rhsm.log"""
+    wait_for(
+        lambda: 'Host-to-guest mapping being sent to' in get_rhsm_log(),
+        timeout=10,
+        delay=2,
+    )
+    logs = get_rhsm_log()
+    print(logs)
```
Comment (shubhamsg199) [inline robottelo/utils/virtwho.py:225]:
```suggestion

```

File: tests/foreman/virtwho/api/test_nutanix.py [hunk 2]
```diff
@@ -295,3 +298,69 @@ def test_positive_prism_central_prism_central_option(
...
+        :id: aa6a7443-62ae-4a87-8c27-4a7808f4b1da308
+
+        :expectedresults:
+            enable-ahv-debug option has been set to no
+            ahv_internal_debug bas been set to false in virt-who-config-X.conf
+            warning message exist in log file /var/log/rhsm/rhsm.log
+            ahv_internal_debug option can be updated
+            message Host UUID {system_uuid} found for VM: {guest_uuid} exist in rhsm.log
+            ahv_internal_debug bas been set to true in virt-who-config-X.conf
+            warning message does not exist in log file /var/log/rhsm/rhsm.log
```
Comment (shubhamsg199) [inline tests/foreman/virtwho/api/test_nutanix.py:317]: Numbering expectedresults would look nice
```suggestion
        :expectedresults:
            1. enable-ahv-debug option has been set to no
            2. ahv_internal_debug bas been set to false in virt-who-config-X.conf
            3. warning message exist in log file /var/log/rhsm/rhsm.log
            4. ahv_internal_debug option can be updated
            5. message Host UUID {system_uuid} found for VM: {guest_uuid} exist in rhsm.log
            6. ahv_internal_debug bas been set to true in virt-who-config-X.conf
            7. warning message does not exist in log file /var/log/rhsm/rhsm.log
```

File: tests/foreman/virtwho/cli/test_nutanix.py [hunk 3]
```diff
@@ -255,3 +258,63 @@ def test_positive_prism_central_prism_central_option(
...
+        :id: a169a843-6ec7-4ef1-b471-649304e77cf1268
+
+        :expectedresults:
+            enable-ahv-debug option has been set to no
+            ahv_internal_debug bas been set to false in virt-who-config-X.conf
+            warning message exist in log file /var/log/rhsm/rhsm.log
+            ahv_internal_debug option can be updated
+            message Host UUID {system_uuid} found for VM: {guest_uuid} exist in rhsm.log
+            ahv_internal_debug bas been set to true in virt-who-config-X.conf
+            warning message does not exist in log file /var/log/rhsm/rhsm.log
```
Comment (shubhamsg199) [inline tests/foreman/virtwho/cli/test_nutanix.py:277]: same numbering the steps can be done as above.

File: tests/foreman/virtwho/ui/test_nutanix.py [hunk 4]
```diff
@@ -237,3 +240,65 @@ def test_positive_prism_central_prism_flavor_option(self, default_org, session,
...
+        :id: af92b814-fe7a-4969-b2ab-7b68859de3ce155
+
+        :expectedresults:
+            enable-ahv-debug option has been set to false
+            ahv_internal_debug bas been set to false in virt-who-config-X.conf
+            warning message exist in log file /var/log/rhsm/rhsm.log
+            ahv_internal_debug option can be updated
+            message Host UUID {system_uuid} found for VM: {guest_uuid} exist in rhsm.log
+            ahv_internal_debug bas been set to true in virt-who-config-X.conf
+            warning message does not exist in log file /var/log/rhsm/rhsm.log
```
Comment (shubhamsg199) [inline tests/foreman/virtwho/ui/test_nutanix.py:257]: same numbering the steps can be done as above.

**Review by shubhamsg199 — CHANGES_REQUESTED** [review]

**Comment by shubhamsg199** [comment]
Please add the z-stream labels if applicable

File: tests/foreman/virtwho/cli/test_nutanix.py (hunk 3 above)
Comment (yanpliu) [reply tests/foreman/virtwho/cli/test_nutanix.py]: > same numbering the steps can be done as above.

@shubhamsg199 Have resolved

File: tests/foreman/virtwho/ui/test_nutanix.py (hunk 4 above)
Comment (yanpliu) [reply tests/foreman/virtwho/ui/test_nutanix.py]: > same numbering the steps can be done as above.

@shubhamsg199 Have updated.

**Comment by yanpliu** [comment]
> requesting

@shubhamsg199 Do you mean add "6.13.z" label? I have done all the updates, please have a review ,thanks

**Comment by JacobCallahan** [comment]
@shubhamsg199 can you revisit this?

File: tests/foreman/virtwho/cli/test_nutanix.py [hunk 5]
```diff
@@ -255,3 +258,63 @@ def test_positive_prism_central_prism_central_option(
...
+            4. ahv_internal_debug option can be updated
+            5. message Host UUID {system_uuid} found for VM: {guest_uuid} exist in rhsm.log
+            6. ahv_internal_debug bas been set to true in virt-who-config-X.conf
+            7. warning message does not exist in log file /var/log/rhsm/rhsm.log
+        :CaseLevel: Integration
+
+        :CaseImportance: Medium
+
+        :BZ: 2141719
+        """
```
Comment (shubhamsg199) [inline tests/foreman/virtwho/cli/test_nutanix.py:283]: I see customer case attached with this BZ
```suggestion
        :customerscenario: true
        """
```

File: tests/foreman/virtwho/ui/test_nutanix.py [hunk 6]
```diff
@@ -237,3 +240,65 @@ def test_positive_prism_central_prism_flavor_option(self, default_org, session,
...
+            4. ahv_internal_debug option can be updated
+            5. message Host UUID {system_uuid} found for VM: {guest_uuid} exist in rhsm.log
+            6. ahv_internal_debug bas been set to true in virt-who-config-X.conf
+            7. warning message does not exist in log file /var/log/rhsm/rhsm.log
+        :CaseLevel: Integration
+
+        :CaseImportance: Medium
+
+        :BZ: 2141719
+        """
```
Comment (shubhamsg199) [inline tests/foreman/virtwho/ui/test_nutanix.py:263]:
```suggestion
        :customerscenario: true
        """
```

**Review by shubhamsg199 — APPROVED** [review]
Looks good, just few changes

**Comment by shubhamsg199** [comment]
> > requesting
> 
> @shubhamsg199 Do you mean add "6.13.z" label? I have done all the updates, please have a review ,thanks

Yes , adding the version labels where you want this PR cherry-picked

**Review by JacobCallahan — APPROVED** [review]

**Pull request merged**